client/components/data/query-reader-teams/index.jsx

```diff
@@ -2,11 +2,14 @@ import { useEffect } from 'react';
 import { useDispatch } from 'react-redux';
 import { requestTeams } from 'calypso/state/teams/actions';
 
-export default function QueryReaderTeams() {
+export function useQueryReaderTeams() {
 	const dispatch = useDispatch();
```

Comment:
I think we can simplify this to 

```js
const dispatch = useDispatch();
dispatch( requestTeams() );
```

Reply:
I don't think so, wouldn't this re-request every render?

Reply:
I think it will do that already as `dispatch` is not stable and will change every render.

Reply:
Yeah, this is how `useDispatch` is created:

```js
export function createDispatchHook(context) {
  if (context === void 0) {
    context = ReactReduxContext;
  }

  var useStore = context === ReactReduxContext ? useDefaultStore : createStoreHook(context);
  return function useDispatch() {
    var store = useStore();
    return store.dispatch;
  };
}
```
So I guess `useEffect` should be removed or should have an empty dependencies array.

Reply:
I read that code and to me it looks like the value of `dispatch` will never change 😅 

> The `dispatch` function reference will be stable as long as the same store instance is being passed to the `<Provider>`. Normally, that store instance never changes in an application.

https://react-redux.js.org/api/hooks#usedispatch

The other reason we want to run it in an effect is it moves it out of the high-priority part of the React lifecycle and into the low-priority part, see https://github.com/Automattic/wp-calypso/pull/72868#discussion_r1098113617

Reply:
> should have an empty dependencies array.

I should also say that this is correct, we don't actually need `dispatch` in the dependencies array because it makes no difference in the way it behaves. It's only there to silence the linter :) Fortunately the linter is clever enough to know that some functions (like a setter returned by `useState()`) doesn't need to go in the dependencies.

Reply:
Yeah, I did read the code closely enough. We create a new function for each render, but its return value is stable.